From 8a69b8c762382f1759f07411e72ecce8cef0432d Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 12 Mar 2014 21:29:01 -0400
Subject: [PATCH] Misc Utils

Adds Misc Util functions to be used by future commits
---
 .../com/empireminecraft/cbmisc/CBMiscUtils.java    | 36 ++++++++++++++++++++++
 .../minecraft/server/GameProfileSerializer.java    |  2 +-
 2 files changed, 37 insertions(+), 1 deletion(-)
 create mode 100644 src/main/java/com/empireminecraft/cbmisc/CBMiscUtils.java

diff --git a/src/main/java/com/empireminecraft/cbmisc/CBMiscUtils.java b/src/main/java/com/empireminecraft/cbmisc/CBMiscUtils.java
new file mode 100644
index 0000000..db73624
--- /dev/null
+++ b/src/main/java/com/empireminecraft/cbmisc/CBMiscUtils.java
@@ -0,0 +1,36 @@
+package com.empireminecraft.cbmisc;
+
+
+import net.minecraft.server.ItemMonsterEgg;
+import net.minecraft.server.MinecraftKey;
+import org.apache.commons.lang.exception.ExceptionUtils;
+import org.bukkit.Bukkit;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.craftbukkit.inventory.CraftItemStack;
+import org.bukkit.entity.EntityType;
+
+
+public final class CBMiscUtils {
+    private CBMiscUtils() {}
+
+    public static EntityType getTypeForSpawnEgg(org.bukkit.inventory.ItemStack bukkitItem) {
+        if (bukkitItem.getType() != Material.MONSTER_EGG) {
+            return null;
+        }
+        MinecraftKey key = ItemMonsterEgg.getKey(CraftItemStack.asNMSCopy(bukkitItem));
+        return key != null ? EntityType.fromName(key.getName()) : null;
+    }
+
+    public static void clearEntityMeta() {
+        for (World world : Bukkit.getWorlds()) {
+            for (org.bukkit.entity.Entity entity : world.getEntities()) {
+                entity.getTempMeta().clear();
+            }
+        }
+    }
+
+    public static String stack() {
+        return ExceptionUtils.getFullStackTrace(new Throwable());
+    }
+}
diff --git a/src/main/java/net/minecraft/server/GameProfileSerializer.java b/src/main/java/net/minecraft/server/GameProfileSerializer.java
index d311e71..d7ef3c6 100644
--- a/src/main/java/net/minecraft/server/GameProfileSerializer.java
+++ b/src/main/java/net/minecraft/server/GameProfileSerializer.java
@@ -220,7 +220,7 @@ public final class GameProfileSerializer {
     }
 
     private static <T extends Comparable<T>> IBlockData a(IBlockData iblockdata, IBlockState<T> iblockstate, String s) {
-        return iblockdata.set(iblockstate, (Comparable) iblockstate.b(s).get());
+        return iblockdata.set(iblockstate, iblockstate.b(s).get()); // EMC - decompile error
     }
 
     public static NBTTagCompound a(NBTTagCompound nbttagcompound, IBlockData iblockdata) {
-- 
2.10.2

